private/highscoreList.js

```diff
@@ -0,0 +1,95 @@
+const router = require('express').Router();
+const db = require('../database/connection.js').connection;
+
+router.get('/api/highscores/user/:userHighestScore', (req, res) => {
+  db.query(`(SELECT high_score.high_score_id, player.username, high_score.score, high_score.date_time 
+        FROM player JOIN high_score 
+        ON player.player_id = high_score.player_id 
+        WHERE score >= ?
+        ORDER BY score ASC
+        LIMIT 5)
+    
+    UNION ALL
+    
+    (SELECT high_score.high_score_id, player.username, high_score.score, high_score.date_time 
+        FROM player JOIN high_score 
+        ON player.player_id = high_score.player_id
+        WHERE score < ?
+        ORDER BY score DESC
+        LIMIT 5);`, [req.params.userHighestScore, req.params.userHighestScore], (error, result, fields) => {
+    if (result && result.length) {
+      const highscores = [];
+      for (const highscore of result) {
+        highscores.push({
+          place: highscore.high_score_id, username: highscore.username, score: highscore.score, dateTime: highscore.date_time,
+        });
+      }
+      res.send({
+        highscores,
+      });
+    } else {
+      res.send({
+        message: 'No scores found',
+      });
+    }
+  });
+});
+
+router.get('/api/highestscore/user/:userId', (req, res) => {
+  db.query('SELECT score FROM high_score WHERE player_id = ? ORDER BY score DESC LIMIT 1;', [req.params.userId], (error, result, fields) => {
+    if (result && result.length) {
+      res.send({
+        highestscore: result[0].score,
+      });
+    } else {
+      res.send({
+        message: 'No scores found.',
+      });
+    }
+  });
+});
+
+router.get('/api/highestscores/:currentPage', (req, res) => {
+  db.query('SELECT * FROM high_score;',(error, result, fields) => {
+    if (result && result.length) {
+      const pageLimit = 1;
+      const currentPage = req.params.currentPage;
+      const scoresSize = result.length;
+      const offset = (currentPage - 1) *pageLimit; //minus one because offset defines from which row we want to retrieve data
+      const fullPages = scoresSize/pageLimit;
+      const reminderOfScores = scoresSize % pageLimit === 0 ? 0 : 1;
+      const pages = fullPages + reminderOfScores;
+      db.query(`SELECT high_score.high_score_id, player.username, high_score.score, high_score.date_time 
+      FROM player 
+      JOIN high_score
+      ON player.player_id = high_score.player_id
+      LIMIT ?
+      OFFSET ?;`, [pageLimit,offset], (error, result2, fields) => {
```

Comment:
Do you need the result2 here? Can't you just use result? As I don't see it being used anywhere after so it doesn't seem like it would matter if it gets overwritten

Reply:
yes yes it was for testing